### PR TITLE
Add testing infrastructure

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require 'rake'
+require 'rake/testtask'
+
+desc 'Test the Foreman Proxy plugin.'
```
Comment (timogoebel) [inline Rakefile]: s/Foreman Proxy plugin./Prprocessor/g

Comment (ekohl) [reply Rakefile]: I may or may not have copy pasted something

File: Gemfile.lock [hunk 2]
```diff
@@ -1,42 +1,55 @@
...
       sawyer (~> 0.6.0, >= 0.5.3)
-    rack (1.5.2)
+    rack (1.6.8)
     rack-protection (1.5.3)
       rack
-    rest_client (1.7.3)
-      netrc (~> 0.7.7)
+    rake (12.2.1)
+    rest-client (1.8.0)
+      http-cookie (>= 1.0.2, < 2.0)
```
Comment (GregSutcliffe) [inline Gemfile.lock]: 1.7 -> 1.8 in rest client is a bit of a risk? IIRC rest client can be a total pain, is this needed *before* we can add tests to verify it works?

Comment (ekohl) [reply Gemfile.lock]: I didn't carefully check and just wanted to pin older than 2.0. Changing to 1.7.x

Comment (GregSutcliffe) [reply Gemfile.lock]: Admittedly this repo is not live, we could merge it and no push to Openshift ...

Comment (ekohl) [reply Gemfile.lock]: Let's keep this in shape to deploy.

**Comment by ekohl** [comment]
Updated with a pin to ```~> 1.7.0```.

**Comment by GregSutcliffe** [comment]
This looks good, I'll merge and deploy to Openshift. Ty @ekohl

**Comment by GregSutcliffe** [comment]
Deployed to openshift

**Pull request merged**